src/js/site/sidebar.test.js

```diff
@@ -0,0 +1,98 @@
+import '../../../jest/global';
+import * as sidebar from './sidebar';
+
+describe('Sidebar', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('highlightTocOnScroll', () => {
+    let h2, tocItem;
+
+    beforeEach(() => {
+      jest.resetAllMocks();
+      h2 = document.createElement('h2');
+      tocItem = document.createElement('a');
+      let queriedTocItems = document.createDocumentFragment();
+      let queriedHeadings = document.createDocumentFragment();
+
+      jest
+        .spyOn(document, 'querySelectorAll')
+        .mockImplementation((selector) => {
+          switch (selector) {
+            case '.toc-entry a':
+              queriedTocItems.appendChild(tocItem);
+              return queriedTocItems.childNodes;
+            case 'h2, h3, h4, h5, h6':
+              queriedHeadings.appendChild(h2);
+              return queriedHeadings.childNodes;
+          }
```

Comment:
probably can do something like this if we want to clean it up a little:

```suggestion
          let items = document.createDocumentFragment();
          switch (selector) {
            case '.toc-entry a':
              items.appendChild(tocItem);
            case 'h2, h3, h4, h5, h6':
              items.appendChild(h2);
          }
          return items;
```

Reply:
I think these need to be in two different node lists (headings and tocItems) if we do this it will put both items in the same list, I think?

Reply:
@teesloane I just pushed a commit with what I had in mind: https://github.com/circleci/circleci-docs/pull/5924/commits/e59da7b3bc28ba3ac8980554c88e73eecc9fd357

Reply:
oh, I see! I misunderstood fragments. Nice!